adam/perception/perception_graph_nodes.py

```diff
@@ -0,0 +1,86 @@
+from abc import ABC
+from typing import Union, Tuple, Any
+
+from attr import attrs, attrib
+from attr.validators import instance_of, in_
+
+from adam.axis import GeonAxis
+from adam.geon import Geon, CrossSection
+from adam.ontology import OntologyNode
+from adam.ontology.phase1_spatial_relations import (
+    Region,
+    SpatialPath,
+    PathOperator,
+    Distance,
+)
+from adam.perception import ObjectPerception
+from adam.semantics import ObjectSemanticNode
+
+
+# Perception graph nodes
+class GraphNode(ABC):
+    """Super-class for all perception graph nodes, useful for types."""
```

Comment:
It could be useful to include optional X, Y information here. Although we won't necessarily use that for patterns, it would be useful when passed on to the UI later.

Reply:
@denizbeser -- Would you think storing one (X,Y) pair as the center of the feature of interest is reasonable enough for now? We may need this to become more robust in the future but I think future robustness is going to be a question of what information we can get about where the features are located from @shengcheng ?

Reply:
Yes, I think so, center should be enough!